ceph.spec.in

```diff
@@ -1399,7 +1399,7 @@ install -m 0644 -D COPYING %{buildroot}%{_docdir}/ceph/COPYING
 install -m 0644 -D etc/sysctl/90-ceph-osd.conf %{buildroot}%{_sysctldir}/90-ceph-osd.conf
 install -m 0755 -D src/tools/rbd_nbd/rbd-nbd_quiesce %{buildroot}%{_libexecdir}/rbd-nbd/rbd-nbd_quiesce
 
-install -m 0755 src/cephadm/cephadm %{buildroot}%{_sbindir}/cephadm
+install -m 0755 src/cephadm/cephadm.py %{buildroot}%{_sbindir}/cephadm
```

Comment:
doesn't really help in splitting cephadm into different python modules. In fact this would actually make it more cumbersome to refactor cephadm. What we need is to introduce a complication step hooked into cmake from a source tree into a cephadm binary, like [so](https://www.google.com/search?q=python+self-contained+executable). 

There are a few places where we really depend on having a single executable: 

https://github.com/ceph/ceph/blob/94224614e0a5b663d394134fadded446b298c8f1/qa/tasks/cephadm.py#L132-L138

https://github.com/ceph/ceph/blob/94224614e0a5b663d394134fadded446b298c8f1/doc/cephadm/install.rst#L60

Renaming cephadm would be a [breaking change](https://www.google.com/search?sa=X&q=%22cephadm%22+%22githubusercontent%22). 

@djgalloway : In order to avoid having users directly downloading the source code, do you think we could somehow put cephadm on download.ceph.com or similar? This would have another great advantage: have proper releases for the cephadm binary. Right now, users directly download the latest cephadm from the stable branch as soon as we merge backport PRs.

Reply:
We can...

Just for my understanding's sake, why isn't it a package of its own?

Reply:
So the thing is: We don't provide official packages for e.g. Debian. Users on those distros rely on the cephadm binary to be easily downloadable.